firestore/query_test.go

```diff
@@ -1770,55 +1775,134 @@ func TestFindNearest(t *testing.T) {
+					Limit:               &wrapperspb.Int32Value{Value: int32(limit)},
+					DistanceMeasure:     pb.StructuredQuery_FindNearest_EUCLIDEAN,
+					DistanceThreshold:   &wrapperspb.DoubleValue{Value: float64(threshold)},
+					DistanceResultField: resultField,
+				},
+			},
+		},
+	}
 
 	testcases := []struct {
-		desc        string
-		path        string
-		queryVector interface{}
-		wantErr     bool
+		desc    string
+		vQuery  VectorQuery
+		wantReq protoreflect.ProtoMessage
+		wantErr error
 	}{
 		{
-			desc:    "Invalid path",
-			path:    "path*",
-			wantErr: true,
+			desc: "Invalid path",
+			vQuery: c.Collection(collName).
+				FindNearest("path*", nil, limit, DistanceMeasureEuclidean, nil),
+			wantErr: errInvalidRunesField("path*"),
+		},
+		{
+			desc: "Invalid vector type",
+			vQuery: c.Collection(collName).
+				FindNearest("path", "abcd", limit, DistanceMeasureEuclidean, nil),
+			wantErr: errInvalidVector,
 		},
 		{
-			desc:        "Valid path",
-			path:        "path",
-			queryVector: []float64{5, 6, 7},
-			wantErr:     false,
+			desc: "Valid path with valid vector type []float64",
+			vQuery: c.Collection(collName).
+				FindNearest("path", []float64{5, 6, 7}, limit, DistanceMeasureEuclidean, nil),
+			wantReq: &wantReq,
 		},
 		{
-			desc:        "Invalid vector type",
-			path:        "path",
-			queryVector: "abcd",
-			wantErr:     true,
+			desc: "Valid path with valid vector type []float32",
+			vQuery: c.Collection(collName).
+				FindNearest("path", []float32{5, 6, 7}, limit, DistanceMeasureEuclidean, nil),
+			wantReq: &wantReq,
 		},
 		{
-			desc:        "Valid vector type",
-			path:        "path",
-			queryVector: []float32{5, 6, 7},
-			wantErr:     false,
+			desc: "Valid path with valid vector type WithDistanceResultField and WithDistanceThreshold ",
+			vQuery: c.Collection(collName).
+				FindNearest("path", []float32{5, 6, 7}, limit, DistanceMeasureEuclidean, &FindNearestOptions{
+					DistanceThreshold:   Ptr[float64](threshold),
+					DistanceResultField: resultField,
+				}),
+			wantReq: &wantReqThresholdField,
 		},
 	}
 	for _, tc := range testcases {
-		srv.reset()
-		srv.addRPC(nil, []interface{}{
-			&pb.RunQueryResponse{Document: wantPBDocs[0]},
+		t.Run(tc.desc, func(t *testing.T) {
+			srv.reset()
+			if tc.wantErr == nil {
+				srv.addRPC(tc.wantReq, []interface{}{
+					&pb.RunQueryResponse{Document: wantPBDocs[0]},
+				})
+			}
+			_, gotErr := tc.vQuery.Documents(ctx).GetAll()
+			if !equalErrs(gotErr, tc.wantErr) {
+				t.Fatalf("got %v, want %v", gotErr, tc.wantErr)
+			}
 		})
-		vQuery := c.Collection("C").FindNearest(tc.path, tc.queryVector, 2, DistanceMeasureEuclidean, nil)
+	}
+}
 
-		_, err := vQuery.Documents(ctx).GetAll()
-		if err == nil && tc.wantErr {
-			t.Fatalf("%s: got nil wanted error", tc.desc)
-		} else if err != nil && !tc.wantErr {
-			t.Fatalf("%s: got %v, want nil", tc.desc, err)
-		}
+func equalErrs(gotErr error, wantErr error) bool {
```

Comment:
Since it's not really equality, I'd rather that `equal` wasn't in the name. Maybe `errorsMatch`?

Reply:
Done